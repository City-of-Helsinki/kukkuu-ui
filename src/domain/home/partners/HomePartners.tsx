import * as React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './homePartners.module.scss';
import Icon from '../../../common/components/icon/Icon';
import amosrexIcon from '../../../assets/icons/partners/amosrex.png';
import cirkoIcon from '../../../assets/icons/partners/cirko.png';
import designmuseumIcon from '../../../assets/icons/partners/designmuseum.png';
import hamhelsinkiIcon from '../../../assets/icons/partners/hamhelsinki.png';
import helIcon from '../../../assets/icons/partners/hel.png';
import helsinginkaupunginmuseoIcon from '../../../assets/icons/partners/helsinginkaupunginmuseo.png';
import helsinginkaupunginorkesteriIcon from '../../../assets/icons/partners/helsinginkaupunginorkesteri.png';
import hktIcon from '../../../assets/icons/partners/hkt.png';
import hotellijaravintolamuseoIcon from '../../../assets/icons/partners/hotellijaravintolamuseo.png';
import hurjaruuthIcon from '../../../assets/icons/partners/hurjaruuth.png';
import jaesIcon from '../../../assets/icons/partners/jaes.png';
import kansallismuseoIcon from '../../../assets/icons/partners/kansallismuseo.png';
import kansallisteatteriIcon from '../../../assets/icons/partners/kansallisteatteri.png';
import kaviIcon from '../../../assets/icons/partners/kavi.png';
import kulttuuriperintokasvatusIcon from '../../../assets/icons/partners/kulttuuriperintokasvatus.png';
import mfaIcon from '../../../assets/icons/partners/mfa.png';
import nukketeatterisampoIcon from '../../../assets/icons/partners/nukketeatterisampo.png';
import oodihelsinkiIcon from '../../../assets/icons/partners/oodihelsinki.png';
import oopperabalettiIcon from '../../../assets/icons/partners/oopperabaletti.png';
import qteatteriIcon from '../../../assets/icons/partners/qteatteri.png';
import svenskateaternIcon from '../../../assets/icons/partners/svenskateatern.png';
import tanssintaloIcon from '../../../assets/icons/partners/tanssintalo.png';
import teatteriilmioIcon from '../../../assets/icons/partners/teatteriilmio.png';
import teatterimuseoIcon from '../../../assets/icons/partners/teatterimuseo.png';
import valokuvataiteenmuseoIcon from '../../../assets/icons/partners/valokuvataiteenmuseo.png';

const HomePartners: React.FunctionComponent = props => {
  const { t } = useTranslation();

  interface Partner {
    name: string;
    icon: string;
  }

  type Props = {
    partners: Partner[];
  };

  // Name, icon file name and translation key is created from partner's domain
  // name without .fi and dash.
  // Examples:
  // teatteri-ilmio.fi -> teatteriilmio
  // hel.fi -> hel
  const mainPartnerList: Partner[] = [
    { name: 'hel', icon: helIcon },
    { name: 'jaes', icon: jaesIcon },
  ];
  const partnerList: Partner[] = [
    { name: 'amosrex', icon: amosrexIcon },
    { name: 'cirko', icon: cirkoIcon },
    { name: 'designmuseum', icon: designmuseumIcon },
    { name: 'hamhelsinki', icon: hamhelsinkiIcon },
    { name: 'helsinginkaupunginmuseo', icon: helsinginkaupunginmuseoIcon },
    {
      name: 'helsinginkaupunginorkesteri',
      icon: helsinginkaupunginorkesteriIcon,
    },
    { name: 'hkt', icon: hktIcon },
    { name: 'hotellijaravintolamuseo', icon: hotellijaravintolamuseoIcon },
    { name: 'hurjaruuth', icon: hurjaruuthIcon },
    { name: 'kansallismuseo', icon: kansallismuseoIcon },
    { name: 'kansallisteatteri', icon: kansallisteatteriIcon },
    { name: 'kavi', icon: kaviIcon },
    { name: 'kulttuuriperintokasvatus', icon: kulttuuriperintokasvatusIcon },
    { name: 'mfa', icon: mfaIcon },
    { name: 'nukketeatterisampo', icon: nukketeatterisampoIcon },
    { name: 'oodihelsinki', icon: oodihelsinkiIcon },
    { name: 'oopperabaletti', icon: oopperabalettiIcon },
    { name: 'qteatteri', icon: qteatteriIcon },
    { name: 'svenskateatern', icon: svenskateaternIcon },
    { name: 'tanssintalo', icon: tanssintaloIcon },
    { name: 'teatteriilmio', icon: teatteriilmioIcon },
    { name: 'teatterimuseo', icon: teatterimuseoIcon },
    { name: 'valokuvataiteenmuseo', icon: valokuvataiteenmuseoIcon },
  ];

  function Partners(props: Props) {
    // any is workaround for TS bug
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    const partners: any = props.partners;
    const partnerItems = partners.map((partner: Partner, index: number) => (
      <Icon
        key={index}
        className={styles.icon}
        src={partner.icon}
        alt={t(`home.partners.partner.${partner.name}`)}
      />
    ));
    return partnerItems;
  }

  return (
    <section className={styles.wrapper}>
      <h2>{t('home.partners.heading.text')}</h2>
      <div className={styles.partners}>
        <div className={styles.big}>
          <Partners partners={mainPartnerList} />
        </div>
        <div className={styles.small}>
          <Partners partners={partnerList} />
          <Icon
            className={styles.icon}
            src={helsinginkaupunginorkesteriIcon}
            alt=""
          />
        </div>
      </div>
    </section>
  );
};

export default HomePartners;
