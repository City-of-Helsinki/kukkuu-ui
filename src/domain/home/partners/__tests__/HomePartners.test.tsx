import { customRender as render } from '../../../../common/test/customRender';
import HomePartners from '../HomePartners';

it('renders snapshot correctly', () => {
  const { container } = render(<HomePartners />);
  expect(container).toMatchSnapshot();
});
