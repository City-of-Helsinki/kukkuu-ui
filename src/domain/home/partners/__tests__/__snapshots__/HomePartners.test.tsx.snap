// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders snapshot correctly 1`] = `
<div>
  <section
    class="_wrapper_1b4993"
  >
    <div
      class="_innerwrapper_1b4993"
    >
      <h2>
        Yhteistyökumppanit
      </h2>
      <div
        class="_partners_1b4993"
      >
        <div
          class="_big_46f5db"
        >
          <a
            class="_container_46f5db"
            href="https://www.hel.fi/fi/paatoksenteko-ja-hallinto/kaupungin-organisaatio/toimialat/kulttuurin-ja-vapaa-ajan-toimiala"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="home.partners.partner.hel"
                src="data:image/svg+xml;base64,PHN2ZyB2aWV3Qm94PSIwIDAgNzggMzYiIHRpdGxlPSJIZWxzaW5naW4ga2F1cHVua2kiIHJvbGU9ImltZyIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICAgIDxwYXRoCiAgICAgICAgZD0iTTc1Ljc1MyAyLjI1MXYyMC43YzAgMy45NS0zLjI3NSA3LjE3OC03LjMxIDcuMTc4aC0yMi4yNmMtMi42NzQgMC01LjIwNS45Ni03LjE4MyAyLjczOWExMC43NDkgMTAuNzQ5IDAgMDAtNy4xODMtMi43NEg5LjUwOWMtNC4wMDMgMC03LjI0Ny0zLjIxLTcuMjQ3LTcuMTc3VjIuMjVoNzMuNDkxek00MC4xODcgMzQuODM1YTguNDcgOC40NyAwIDAxNi4wMTItMi40NzFoMjIuMjQ1YzUuMjY4IDAgOS41NTYtNC4yMTkgOS41NTYtOS40MTNWMEgwdjIyLjkzNWMwIDUuMTk0IDQuMjU2IDkuNDEzIDkuNTA5IDkuNDEzaDIyLjMwOGMyLjI2MyAwIDQuMzk4Ljg4MiA2LjAxMiAyLjQ3MUwzOS4wMTYgMzZsMS4xNy0xLjE2NXoiCiAgICAgICAgZmlsbD0iY3VycmVudENvbG9yIiAvPgogICAgPHBhdGgKICAgICAgICBkPSJNNjcuNTIyIDExLjY3NmMwIC42ODEtLjU1NiAxLjE3Ny0xLjI1NSAxLjE3Ny0uNyAwLTEuMjU1LS40OTYtMS4yNTUtMS4xNzcgMC0uNjgyLjU1Ni0xLjE3OCAxLjI1NS0xLjE3OC43LS4wMyAxLjI1NS40NjUgMS4yNTUgMS4xNzh6bS0yLjM1MiA5LjYyMmgyLjE3OHYtNy41NDZINjUuMTd2Ny41NDZ6bS0zLjkwOS00LjU1NmwyLjg0NSA0LjU1NmgtMi4zNjhsLTEuOTA3LTMuMDIyLTEuMDMzIDEuMjcxdjEuNzVoLTIuMTYxVjEwLjQ1M2gyLjE2djUuMDA0YzAgLjkzLS4xMSAxLjg2LS4xMSAxLjg2aC4wNDdzLjUwOS0uODIxLjkzOC0xLjQxbDEuNjUzLTIuMTU0aDIuNTQybC0yLjYwNiAyLjk5em0tNi44MTctLjI3OGMwLTEuODc1LS45MzgtMi44OTgtMi40MzItMi44OTgtMS4yNzEgMC0xLjkzOS43MjgtMi4zMiAxLjQyNmgtLjA0OGwuMTEyLTEuMjRoLTIuMTYydjcuNTQ2aDIuMTYyVjE2LjgyYzAtLjg2OC41MjQtMS40NzIgMS4zMzUtMS40NzIuODEgMCAxLjE2LjUyNyAxLjE2IDEuNTM0djQuNDE2aDIuMTc3bC4wMTYtNC44MzR6bS04LjkzMS00Ljc4OGMwIC42ODEtLjU1NyAxLjE3Ny0xLjI1NiAxLjE3Ny0uNyAwLTEuMjU1LS40OTYtMS4yNTUtMS4xNzcgMC0uNjgyLjU1Ni0xLjE3OCAxLjI1NS0xLjE3OC43MTUtLjAzIDEuMjU2LjQ2NSAxLjI1NiAxLjE3OHptLTIuMzUyIDkuNjIyaDIuMTc3di03LjU0Nkg0My4xNnY3LjU0NnptLTMuNzUtMi4xMDdjMC0uNjA1LS44NTktLjcyOS0xLjg2LTEuMDA4LTEuMTYtLjI5NC0yLjYyMi0uODY3LTIuNjIyLTIuMzA4IDAtMS40MjYgMS4zOTgtMi4zMjQgMy4wNTEtMi4zMjQgMS41NDEgMCAyLjk1Ni43MTIgMy41NDQgMS43MmwtMS44NiAxLjAyMmMtLjE5LS42NjYtLjc2Mi0xLjE5My0xLjYyLTEuMTkzLS41NTcgMC0xLjAxOC4yMzItMS4wMTguNjgyIDAgLjU3MyAxLjAxOC42MzUgMi4xNjIuOTkxIDEuMjA4LjM3MiAyLjMyLjkxNSAyLjMyIDIuMjk0IDAgMS41MTgtMS40NDYgMi40MTctMy4xMTUgMi40MTctMS44MTEgMC0zLjI0Mi0uNzQ0LTMuODc3LTEuOTUybDEuODktMS4wMzljLjI0LjgyMi45MjIgMS40NDEgMS45NTUgMS40NDEuNjIgMCAxLjA1LS4yNDggMS4wNS0uNzQzem0tNi44ODItOC42NzdoLTIuMTc3djguNjkyYzAgLjc3NS4xNzUgMS4zNDguNTA5IDEuNzA1LjM1LjM1Ni44OS41MjYgMS42MzYuNTI2LjI1NSAwIC41MjUtLjAzLjc4LS4wNzcuMjctLjA2Mi40NzYtLjE0LjY1LS4yMzNsLjE5MS0xLjQyNWEyLjA3IDIuMDcgMCAwMS0uNDYuMTI0Yy0uMTI4LjAzLS4yODcuMDMtLjQ2MS4wMy0uMjg2IDAtLjQxNC0uMDc3LS41MDktLjIxNi0uMTExLS4xNC0uMTU5LS4zODctLjE1OS0uNzQ0di04LjM4MnptLTcuMjQ2IDQuNTdjLS43OTUgMC0xLjQ0Ni41NTgtMS42MjEgMS41ODFoMy4wNWMuMDE3LS44OTktLjU4Ny0xLjU4LTEuNDMtMS41OHptMy4zNTMgMy4wMDdIMjMuNjNjLjA5NSAxLjIyNC43OTQgMS44MjggMS43IDEuODI4LjgxIDAgMS4zNjctLjUyNyAxLjQ5NC0xLjI0bDEuODI4IDEuMDA3Yy0uNTQuOTYxLTEuNyAxLjc5OC0zLjMyMiAxLjc5OC0yLjE2IDAtMy43NS0xLjQ3Mi0zLjc1LTMuOTUxIDAtMi40NjQgMS42Mi0zLjk1MSAzLjcwMy0zLjk1MSAyLjA4MSAwIDMuNDY0IDEuNDQgMy40NjQgMy40ODYtLjAxNi42MDQtLjExMSAxLjAyMy0uMTExIDEuMDIzem0tMTEuMDc3IDMuMjA3aDIuMjU3VjEwLjkxNmgtMi4yNTd2NC4xMDdoLTQuMjQzdi00LjA5MUgxMS4wNnYxMC4zNjZoMi4yNTZ2LTQuMjkyaDQuMjQzdjQuMjkyeiIKICAgICAgICBmaWxsPSJjdXJyZW50Q29sb3IiIC8+Cjwvc3ZnPg=="
              />
            </div>
          </a>
          <a
            class="_container_46f5db _jaes-icon_46f5db"
            href="https://jaes.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Jane ja Aatos Erkon säätiö"
                src="/src/assets/icons/partners/jaes-logo-fi.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="_small_46f5db"
        >
          <a
            class="_container_46f5db"
            href="https://amosrex.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Amos Rex"
                src="/src/assets/icons/partners/amosrex.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://cirko.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Cirko – Uuden sirkuksen keskus"
                src="/src/assets/icons/partners/cirko.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://dotdot.fi/suomeksi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="DOT ry"
                src="/src/assets/icons/partners/dotdot.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.designmuseum.fi/fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Designmuseo"
                src="/src/assets/icons/partners/designmuseum.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.hamhelsinki.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Helsingin taidemuseo HAM"
                src="/src/assets/icons/partners/hamhelsinki.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.helsinginkaupunginmuseo.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Helsingin kaupunginmuseo"
                src="/src/assets/icons/partners/helsinginkaupunginmuseo.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://helsinginkaupunginorkesteri.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Helsingin kaupunginorkesteri"
                src="/src/assets/icons/partners/helsinginkaupunginorkesteri.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://hkt.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Helsingin Kaupunginteatteri"
                src="/src/assets/icons/partners/hkt.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.hotellijaravintolamuseo.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Hotelli- ja ravintolamuseo"
                src="/src/assets/icons/partners/hotellijaravintolamuseo.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.hurjaruuth.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Tanssiteatteri Hurjaruuth"
                src="/src/assets/icons/partners/hurjaruuth.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.kaapelitehdas.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="home.partners.partner.kaapeli"
                src="/src/assets/icons/partners/Kaapeli.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.kansallisgalleria.fi/fi/search?category=artwork&hasImage=true"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="home.partners.partner.kansallisgalleria"
                src="/src/assets/icons/partners/Finnish_National_Gallery_Logo_FI.svg"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.kansallismuseo.fi/fi/kansallismuseo"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Kansallismuseo"
                src="/src/assets/icons/partners/kansallismuseo.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://kansallisteatteri.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Kansallisteatteri"
                src="/src/assets/icons/partners/kansallisteatteri.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://kavi.fi/kansallinen-audiovisuaalinen-instituutti-palvelee/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Kansallinen audiovisuaalinen instituutti"
                src="/src/assets/icons/partners/kavi.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.kulttuuriperintokasvatus.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Suomen Kulttuuriperintökasvatuksen seura"
                src="/src/assets/icons/partners/kulttuuriperintokasvatus.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.helsinki.fi/fi/tiedemuseo-liekki"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Tiedemuseo Liekki"
                src="/src/assets/icons/partners/Museum_Flame_FI.svg"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.mfa.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Arkkitehtuurimuseo"
                src="/src/assets/icons/partners/mfa.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://nukketeatterisampo.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Nukketeatteri Sampo"
                src="/src/assets/icons/partners/nukketeatterisampo.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://oopperabaletti.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Suomen kansallisooppera ja -baletti"
                src="/src/assets/icons/partners/oopperabaletti.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.q-teatteri.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Q-teatteri"
                src="/src/assets/icons/partners/qteatteri.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://svenskateatern.fi/fi/alku/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Svenska Teatern"
                src="/src/assets/icons/partners/svenskateatern.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.tanssintalo.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Tanssin talo"
                src="/src/assets/icons/partners/tanssintalo.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.teatteri-ilmio.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Teatteri ILMI Ö"
                src="/src/assets/icons/partners/teatteriilmio.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.teatterimuseo.fi/"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Teatterimuseo"
                src="/src/assets/icons/partners/teatterimuseo.png"
              />
            </div>
          </a>
          <a
            class="_container_46f5db"
            href="https://www.valokuvataiteenmuseo.fi/fi"
          >
            <div
              class="_imgWrapper_73ff55 _icon_46f5db"
            >
              <img
                alt="Suomen valokuvataiteen museo"
                src="/src/assets/icons/partners/valokuvataiteenmuseo.png"
              />
            </div>
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
